jvm/src/test/scala/fm/common/TestClassUtil.scala

```diff
@@ -0,0 +1,163 @@
+    testFiles.foreach { ClassUtil.classpathFileExists(_) should equal(true) }
+  }
+
+  // classpathFileExists
+  /*
+  // Directories timestamps get changed every time they get moved to the new resource directory/jar file/etc,
+  // This was used to do a manual test for directories, but commenting out to get tests to pass
+  test("classpathLastModified - directories") {
+    // This is the project-relative path for the directory, and test assumes being ran in the project home
+    val f: File = new File("jvm/src/test/resources/test/classutil")
+    assert(f.isDirectory, s"$f must be a directory (is the working directory the project home?)")
+
+    testDirPaths.foreach{ ClassUtil.classpathLastModified(_) should equal(f.lastModified) }
+    testDirFiles.foreach{ ClassUtil.classpathLastModified(_) should equal(f.lastModified) }
+  }*/
+
+  test("classpathLastModified - files") {
+    val f: File = new File(s"jvm/src/test/resources/$testPath")
+    assert(f.isFile, s"$f must be a file (is the working directory the project home?)")
+
+    testPaths.foreach { ClassUtil.classpathLastModified(_) should equal(f.lastModified) }
+    testFiles.foreach { ClassUtil.classpathLastModified(_) should equal(f.lastModified) }
+  }
+
+  test("findAnnotatedClasses") {
+    ClassUtil.findAnnotatedClasses("test.classutil", classOf[java.lang.Deprecated]) should equal(Set(classOf[_root_.test.classutil.TestJavaAnnotatedClass]))
+  }
+
+  test("findClassNames") {
+    ClassUtil.findClassNames("test.classutil") should equal(testClasses)
+  }
+
+  test("findClasspathFiles") {
+    // Includes recursive file(s)
+    val expectedFiles: Set[File] = {
+      testClasses.map{ _.replace(".", "/") + ".class" } ++ Set(testPath, "test/classutil/subdirectory/subfile.txt")
+    }.map{ new File(_) }
+
+    ClassUtil.findClasspathFiles("test.classutil") should equal (expectedFiles)
+  }
+
+  test("findImplementingClasses") {
+    ClassUtil.findImplementingClasses("test.classutil", classOf[_root_.test.classutil.TestTrait]) should equal(Set(classOf[_root_.test.classutil.TestClassExtendsTestTrait]))
+  }
+
+  /*
+    test("def findLoadedClass(cls: String, classLoader: ClassLoader = defaultClassLoader): Option[Class[_]]") { }
+
+    test("def isClassLoaded(cls: String, classLoader: ClassLoader = defaultClassLoader): Boolean") { }
+  */
+
+  test("listClasspathFiles - defaultClassLoader") {
+    // Does NOT include recursive file(s)
+    val updatedTestClasses: Set[String] = testClasses - "test.classutil.subpackage.TestSubPackageClass" // don't include subpackage class
+
+    val expectedFiles: Set[File] = {
+      updatedTestClasses.map{ _.replace(".", "/") + ".class" } ++ Set(testPath, "test/classutil/subpackage", "test/classutil/subdirectory")
+    }.map{ new File(_) }
+
+    ClassUtil.listClasspathFiles("test.classutil") should equal(expectedFiles)
+  }
+
+  test("""listClasspathFiles("")""") {
+    assertThrows[IllegalArgumentException] {
+      ClassUtil.listClasspathFiles("")
```

Comment:
I'm sorta looking into it...I think the implementation is broken for the `listClasspathFiles`, because the `findClasspathFiles` does "work" in one of these situations.